storage/aptosdb/src/lib.rs

```diff
@@ -652,9 +652,16 @@ impl AptosDB {
                 .with_label_values(&["save_transactions_state"])
                 .start_timer();
 
+            let empty = HashMap::new();
             let state_updates_vec = txns_to_commit
                 .iter()
-                .map(|txn_to_commit| txn_to_commit.state_updates())
+                .map(|txn_to_commit| {
+                    if txn_to_commit.is_state_checkpoint() {
+                        &empty
+                    } else {
+                        txn_to_commit.state_updates()
```

Comment:
@msmouse objection? they are all gonna be changed after lazy commit so I think it is okay for now.